### PR TITLE
A small fix in unit tests

### PR DESCRIPTION
**Pull request by CyberDem0n**
Not all external resources were properly mocked

### CODE REVIEW

File: tests/test_patroni.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def test_no_config(self):
         self.assertRaises(SystemExit, patroni_main)
 
     @patch('sys.argv', ['patroni.py', '--validate-config', 'postgres0.yml'])
+    @patch('socket.socket.connect_ex', Mock(return_value=1))
```
Comment (hughcapet) [inline tests/test_patroni.py:57]: [Off-topic] I am wondering, is validate_host_port() with arg listen=False covered anyhow? 

https://github.com/zalando/patroni/blob/641647eeacbcdfeaba91aa48631d2018d8d06176/patroni/validator.py#L48-L52

Comment (CyberDem0n) [reply tests/test_patroni.py:57]: They are (according to coverage).
And the `connect_ex()` is mocked here: https://github.com/zalando/patroni/blob/db9b5962ec5b921c4c98823f623084f6be58b648/tests/test_validator.py#L111-L131

Comment (hughcapet) [reply tests/test_patroni.py:57]: oh, thanks. Missed that part

File: tests/test_consul.py [hunk 2]
```diff
@@ -217,6 +218,7 @@ def test_update_service(self):
         d['role'] = 'bla'
         self.assertIsNone(self.c.update_service({}, d))
 
+    @patch.object(consul.Consul.KV, 'put', Mock(side_effect=ConsulException))
```
[the remark this answers is not in the document]
Comment (CyberDem0n) [reply tests/test_consul.py:221]: Further down it calls `touch_member()` in order to make sure that new settings are applied (service should be deregistered)

File: tests/test_patroni.py (hunk 1 above)
Comment (hughcapet) [reply tests/test_patroni.py:57]: although I still don't see any explicit listen=False setting

**Comment by hughcapet** [comment]
👍

File: tests/test_patroni.py (hunk 1 above)
Comment (CyberDem0n) [reply tests/test_patroni.py:57]: `False` is a default value.
Calls happen from here: https://github.com/zalando/patroni/blob/db9b5962ec5b921c4c98823f623084f6be58b648/patroni/validator.py#L60-L66

Comment (hughcapet) [reply tests/test_patroni.py:57]: 🤦‍♀️ sure, thanks again

**Comment by idanovinda** [comment]
👍

**Pull request merged**